src/cloud/components/Views/index.tsx

```diff
@@ -20,10 +23,21 @@ const ViewsList = ({
   docs,
   currentUserIsCoreMember,
 }: ViewsListProps) => {
+  const targetIdRef = useRef(parent.target.id)
   const [selectedViewId, setSelectedViewId] = useState<number | undefined>(() =>
     views.length > 0 ? views[0].id : undefined
   )
   const { createViewApi } = useCloudApi()
+  const { openModal } = useModal()
+
+  useEffect(() => {
+    if (parent.target.id === targetIdRef.current) {
```

Comment:
What is parent? and what is parent.target?

Reply:
views can belong to either a folder, or a dashboard/smart view

so the parent is just a way to call the correct apis

```
export type ViewParent =
  | { type: 'folder'; target: SerializedFolderWithBookmark }
  | { type: 'smartView'; target: SerializedSmartView }
```